python/seldon_core/utils.py

```diff
@@ -0,0 +1,377 @@
+
+
+def json_to_feedback(message_json: Dict) -> prediction_pb2.Feedback:
+    """
+    Parse a JSON message to a Feedback proto
+    Parameters
+    ----------
+    message_json
+       Input json message
+    Returns
+    -------
+       A SeldonMessage
+    """
+    message_proto = prediction_pb2.Feedback()
+    try:
+        json_format.ParseDict(message_json, message_proto)
+        return message_proto
+    except json_format.ParseError as pbExc:
+        raise SeldonMicroserviceException("Invalid JSON: "+str(pbExc))
+
+
+def json_to_seldon_messages(message_json: Dict) -> prediction_pb2.SeldonMessageList:
+    message_proto = prediction_pb2.SeldonMessageList()
+    try:
+        json_format.ParseDict(message_json, message_proto)
+        return message_proto
+    except json_format.ParseError as pbExc:
+        raise SeldonMicroserviceException("Invalid JSON: "+str(pbExc))
+
+
+def seldon_message_to_json(message_proto: prediction_pb2.SeldonMessage) -> Dict:
+    """
+    Convert a SeldonMessage proto to JSON Dict
+    Parameters
+    ----------
+    message_proto
+       SeldonMessage proto
+    Returns
+    -------
+       JSON Dict
+    """
+    message_json = json_format.MessageToJson(message_proto)
+    message_dict = json.loads(message_json)
+    return message_dict
+
+def seldon_messages_to_json(message_protos: prediction_pb2.SeldonMessageList) -> Dict:
+    """
+    Convert a SeldonMessage proto list to JSON Dict
+    Parameters
+    ----------
+    message_protos
+       SeldonMessage protos
+    Returns
+    -------
+       JSON Dict
+    """
+    message_json = json_format.MessageToJson(message_protos)
+    message_dict = json.loads(message_json)
+    return message_dict
+
+def feedback_to_json(message_proto: prediction_pb2.Feedback) -> Dict:
+    """
+    Convert a SeldonMessage proto to JSON Dict
+    Parameters
+    ----------
+    message_proto
+       SeldonMessage proto
+    Returns
+    -------
+       JSON Dict
+    """
+    message_json = json_format.MessageToJson(message_proto)
+    message_dict = json.loads(message_json)
+    return message_dict
+
+def get_data_from_proto(request: prediction_pb2.SeldonMessage) -> Union[np.ndarray,str,bytes]:
+    """
+    Extract the data payload from the SeldonMessage
+    Parameters
+    ----------
+    request
+       SeldonMessage
+
+    Returns
+    -------
+       Data payload as numpy array or the raw message format. Numpy array will be returned if the "data" field was used.
+
+    """
+    data_type = request.WhichOneof("data_oneof")
+    if data_type == "data":
+        datadef = request.data
+        return grpc_datadef_to_array(datadef)
+    elif data_type == "binData":
+        return request.binData
+    elif data_type == "strData":
+        return request.strData
+    else:
+        raise SeldonMicroserviceException("Unknown data in SeldonMessage")
+
+
+def grpc_datadef_to_array(datadef: prediction_pb2.DefaultData) -> np.ndarray:
```

Comment:
Maybe worth to extract GRPC-related methods to a separate module `utils_grpc.py`.

Reply:
future PR